src/rust/src/demuxer/common_types.rs

```diff
@@ -0,0 +1,363 @@
+use crate::bindings::{lib_ccx_ctx, list_head};
+use lib_ccxr::common::{Codec, Decoder608Report, DecoderDtvccReport, StreamMode, StreamType};
+use lib_ccxr::time::Timestamp;
+use std::ptr::null_mut;
+
+// Size of the Startbytes Array in CcxDemuxer - const 1MB
+pub(crate) const ARRAY_SIZE: usize = 1024 * 1024;
+
+// Constants for Report Information
+pub const SUB_STREAMS_CNT: usize = 10;
+pub const MAX_PID: usize = 65536;
+pub const MAX_PSI_PID: usize = 8191;
+pub const MAX_PROGRAM: usize = 128;
+pub const MAX_PROGRAM_NAME_LEN: usize = 128;
+pub const STARTBYTESLENGTH: usize = 1024 * 1024;
+pub const CCX_NOPTS: i64 = 0x8000_0000_0000_0000u64 as i64;
+
+#[repr(u32)]
+pub enum Stream_Type {
+    PrivateStream1 = 0,
+    Audio,
+    Video,
+    Count,
+}
+#[derive(Debug)]
+pub struct CcxStreamMp4Box {
+    pub box_type: [u8; 4],
+    pub score: i32,
+}
+#[derive(Clone)]
+pub struct CcxDemuxReport {
+    pub program_cnt: u32,
+    pub dvb_sub_pid: [u32; SUB_STREAMS_CNT],
+    pub tlt_sub_pid: [u32; SUB_STREAMS_CNT],
+    pub mp4_cc_track_cnt: u32,
+}
+#[derive(Debug)]
+pub struct FileReport {
+    pub width: u32,
+    pub height: u32,
+    pub aspect_ratio: u32,
+    pub frame_rate: u32,
+    pub data_from_608: Option<Decoder608Report>,
+    pub data_from_708: Option<DecoderDtvccReport>,
+    pub mp4_cc_track_cnt: u32,
+}
+// program_info Struct
+#[derive(Copy, Clone)]
+pub struct ProgramInfo {
+    pub pid: i32,
+    pub program_number: i32,
+    pub initialized_ocr: bool, // Avoid initializing the OCR more than once
+    pub analysed_pmt_once: u8, // 1-bit field
+    pub version: u8,
+    pub saved_section: [u8; 1021],
```

Comment:
No magic numbers, we can take out 1021 as a constant, since it is not something that's general.